### PR TITLE
fixes #21

### LINKED ISSUE #21: Price increase status on UI
Price increase status (Prices go up after ...) on UI should be based on if a higher price is present for the item.

### PR DESCRIPTION
**Pull request by shreyas-satish**

### CODE REVIEW

File: boxoffice/models/item.py [hunk 1]
```diff
@@ -43,6 +43,12 @@ def current_price(self):
         """
         return self.price_at(datetime.utcnow())
 
+    def has_higher_price(self, current_price):
```
Comment (jace) [inline boxoffice/models/item.py]: Not an immediate priority, but all these queries to render a widget are expensive. This value can be cached for up to a week (in our current use), so it needs to be pushed to cache for whatever timeline is left for validity.

I suggest model-level methods should return status and valid_until, and a wrapping function in a workflow object should push this to cache for the valid_until period.

File: boxoffice/templates/boxoffice.html [hunk 2]
```diff
@@ -69,7 +69,14 @@ <h1 class="category-heading">{{title}}</h1>
                           {{/}}
                         </div>
                         {{#if line_item.quantity_available > 0}}
-                          <p class="valid-upto"><i class="fa fa-fw fa-long-arrow-up"></i>Price goes up after {{line_item.price_valid_upto}}</p>
+                          <p class="valid-upto">
+                            {{#if line_item.has_higher_price}}
+                              <i class="fa fa-fw fa-long-arrow-up"></i>
+                              Price goes up after {{line_item.price_valid_upto}}
```
Comment (jace) [inline boxoffice/templates/boxoffice.html]: Isn't `valid_upto` the non-inclusive endpoint, such as "midnight on Saturday"? In that case the phrasing should be "goes up at", not "goes up after".

File: boxoffice/templates/boxoffice.html [hunk 3]
```diff
@@ -69,7 +69,14 @@ <h1 class="category-heading">{{title}}</h1>
...
+                            {{else}}
+                              Sale closes on {{line_item.price_valid_upto}}
```
Comment (jace) [inline boxoffice/templates/boxoffice.html]: Also "at" not "on".

File: requirements.txt [hunk 4]
```diff
@@ -2,7 +2,7 @@ Flask
 Flask-SQLAlchemy
 Flask-Assets
 Flask-Mail
-Flask-WTF
+Flask-WTF==0.12
```
Comment (jace) [inline requirements.txt]: What broke?

File: boxoffice/templates/boxoffice.html (hunk 2 above)
Comment (shreyas-satish) [inline boxoffice/templates/boxoffice.html]: `line_item.price_valid_upto` is formatted as a date, eg: Jan 16 2017. Okay to stick with "goes up after".

File: boxoffice/templates/boxoffice.html (hunk 3 above)
Comment (shreyas-satish) [inline boxoffice/templates/boxoffice.html]: `price_valid_upto` is formatted as a date. "on" works?

File: requirements.txt (hunk 4 above)
Comment (shreyas-satish) [inline requirements.txt]: The latest [version](https://flask-wtf.readthedocs.io/en/stable/changelog.html), seems to have changed the API to selectively disable CSRF token checks.

Comment (jace) [inline requirements.txt]: Let's do a site-wide upgrade then. Can't risk getting bogged down with an old release.

File: boxoffice/templates/boxoffice.html (hunk 3 above)
Comment (jace) [inline boxoffice/templates/boxoffice.html]: "on" works then. You'll need a better way to deal with such nuances though. I don't see any formatting options here in the template, where you'd expect to choose the formatting.

File: boxoffice/templates/boxoffice.html (hunk 2 above)
Comment (jace) [inline boxoffice/templates/boxoffice.html]: Isn't that still "goes up on"? We shouldn't be picking 11:59 PM as end-point. It should be a non-inclusive endpoint at midnight.

**Pull request merged**